src/mscorlib/src/System/Environment.cs

```diff
@@ -1052,8 +1086,27 @@ public static IDictionary GetEnvironmentVariables()
                 root[0]++;
             }
             return result;
+#else // PLATFORM_UNIX
+            // Use reflection to access the unix implementation in System.IO.FileSystem.DriveInfo.dll
+            Type driveInfoType = Type.GetType(
+                "System.IO.DriveInfo, System.IO.FileSystem.DriveInfo, Version=4.0.0.0, Culture=neutral, PublicKeyToken=" + AssemblyRef.MicrosoftPublicKeyToken,
```

Comment:
If the implementation of Environment class lived in corefx repo, would you use reflection to implement this as well?